flocker/control/_config.py

```diff
@@ -1014,9 +1022,15 @@ def _parse_link_configuration(self, application_name, config):
                     raise ValueError(
                         "Unrecognised keys: {keys}.".format(
                             keys=', '.join(sorted(link))))
-                links.append(Link(local_port=local_port,
-                                  remote_port=remote_port,
-                                  alias=alias))
+                try:
+                    links.append(Link(local_port=local_port,
+                                      remote_port=remote_port,
+                                      alias=alias))
+                except InvariantException:
+                    # We've already verified types of Link parameters, so
+                    # that just leaves the alias naming limits as the
+                    # source of error.
```

Comment:
It seemed a little tricky. My *hope* is that we just throw this code away. Reality is maybe we won't for a while... but if we add new invariants presumably we'll have tests for them which will catch the problem? I really don't want to spend too much time thinking about this particular pile of code.